packages/state-transition/src/epoch/processPendingDeposits.ts

```diff
@@ -0,0 +1,117 @@
+    ) {
+      break;
+    }
+
+    // Check if deposit has been finalized, otherwise, stop processing.
+    if (deposit.slot > finalizedSlot) {
+      break;
+    }
+
+    // Check if number of processed deposits has not reached the limit, otherwise, stop processing.
+    if (nextDepositIndex >= MAX_PENDING_DEPOSITS_PER_EPOCH) {
+      break;
+    }
+
+    // Read validator state
+    let isValidatorExited = false;
+    let isValidatorWithdrawn = false;
+
+    const validatorIndex = state.epochCtx.getValidatorIndex(deposit.pubkey);
+    if (validatorIndex !== null) {
+      const validator = state.validators.getReadonly(validatorIndex);
+      isValidatorExited = validator.exitEpoch < FAR_FUTURE_EPOCH;
+      isValidatorWithdrawn = validator.withdrawableEpoch < nextEpoch;
+    }
+
+    if (isValidatorWithdrawn) {
+      // Deposited balance will never become active. Increase balance but do not consume churn
+      applyPendingDeposit(state, deposit, cache);
+    } else if (isValidatorExited) {
+      // Validator is exiting, postpone the deposit until after withdrawable epoch
+      depositsToPostpone.push(deposit);
+    } else {
+      // Check if deposit fits in the churn, otherwise, do no more deposit processing in this epoch.
+      isChurnLimitReached = processedAmount + deposit.amount > availableForProcessing;
+      if (isChurnLimitReached) {
+        break;
+      }
+      // Consume churn and apply deposit.
+      processedAmount += deposit.amount;
+      applyPendingDeposit(state, deposit, cache);
+    }
+
+    // Regardless of how the deposit was handled, we move on in the queue.
+    nextDepositIndex++;
+  }
+
+  const remainingPendingDeposits = state.pendingDeposits.sliceFrom(nextDepositIndex);
+  state.pendingDeposits = remainingPendingDeposits;
+
+  // TODO Electra: add a function in ListCompositeTreeView to support batch push operation
+  for (const deposit of depositsToPostpone) {
+    state.pendingDeposits.push(deposit);
+  }
+
+  // Accumulate churn only if the churn limit has been hit.
+  if (isChurnLimitReached) {
+    state.depositBalanceToConsume = availableForProcessing - BigInt(processedAmount);
+  } else {
+    state.depositBalanceToConsume = 0n;
+  }
+}
+
+function applyPendingDeposit(
+  state: CachedBeaconStateElectra,
+  deposit: PendingDeposit,
+  cache: EpochTransitionCache
+): void {
+  const validatorIndex = state.epochCtx.getValidatorIndex(deposit.pubkey);
+  const {pubkey, withdrawalCredentials, amount, signature} = deposit;
+  const cachedBalances = cache.balances;
+
+  if (validatorIndex === null) {
+    // Verify the deposit signature (proof of possession) which is not checked by the deposit contract
+    if (isValidDepositSignature(state.config, pubkey, withdrawalCredentials, amount, signature)) {
+      addValidatorToRegistry(ForkSeq.electra, state, pubkey, withdrawalCredentials, amount);
+    }
+  } else {
+    // Increase balance
+    increaseBalance(state, validatorIndex, deposit.amount);
```

Comment:
we have `amount` variable already
```suggestion
    increaseBalance(state, validatorIndex, amount);
```